### DISCUSSION TITLE
Support for extra paramaters.

### DISCUSSION THREAD

**Discussion by uzhao** [cline/cline#1047] [Feature Requests]
o1 may need to adjust max_completion_tokens which impact quality of return directly.
deepseek-v3 also have max_tokens which by default it 4k, people may need a larger value in some cases.
So if cline can let user put customer extra paramaters it will be very helpful

**Comment by animikhaich** [comment]
+1 

I keep hitting the Antrhopic rate limit when I should not be hitting the max token limit. I'd like the option to set the max_tokens parameter. 

```
429 {"type":"error","error":{"type":"rate_limit_error","message":"This request would exceed your organization’s rate limit of 8,000 output tokens per minute. For details, refer to: https://docs.anthropic.com/en/api/rate-limits; see the response headers for current usage. Please reduce the prompt length or the maximum tokens requested, or try again later. You may also contact sales at https://www.anthropic.com/contact-sales to discuss your options for a rate limit increase."}}
```

It is almost impossible to use Anthropic API with Cline.
